src/multi.rs

```diff
@@ -527,6 +519,14 @@ enum InsertLocation {
     Before(usize),
 }
 
+// Calculate real length based on terminal width
+// This take in account linewrap from terminal
+fn real_line_count<StrRef: AsRef<str>>(lines: &[StrRef], width: f64) -> usize {
```

Comment:
I took some liberties with the name and docstring -- `count_lines` doesn't say why you'd want this function instead of `.len()`. I picked `visual_line_count`, because I agree `real` isn't good.